lib/vast/Conversion/Common/IRsToLLVM.cpp

```diff
@@ -763,10 +695,16 @@ namespace vast
 
             target.addLegalOp< hl::TypeDefOp >();
 
-            target.addDynamicallyLegalOp< core::LazyOp >(has_llvm_return_type< core::LazyOp >);
-            target.addDynamicallyLegalOp< core::BinLAndOp >(has_llvm_return_type< core::BinLAndOp >);
-            target.addDynamicallyLegalOp< core::BinLOrOp >(has_llvm_return_type< core::BinLOrOp >);
-            target.addDynamicallyLegalOp< hl::ValueYieldOp >(has_llvm_return_type< hl::ValueYieldOp >);
+            auto illegal_with_llvm_ret_type = [&]< typename T >( T && )
+            {
+                target.addDynamicallyLegalOp< T >( has_llvm_return_type< T > );
+            };
+
+            illegal_with_llvm_ret_type( core::LazyOp{} );
+            illegal_with_llvm_ret_type( core::BinLAndOp{} );
+            illegal_with_llvm_ret_type( core::BinLOrOp{} );
+            illegal_with_llvm_ret_type( hl::ValueYieldOp{} );
```

Comment:
Why `illegal`? Shouldn't it be `legal_with_llvm_ret_type`? We are allowing the op only if the return type is in llvm.
```suggestion
            auto legal_with_llvm_ret_type = [&]< typename T >( T && )
            {
                target.addDynamicallyLegalOp< T >( has_llvm_return_type< T > );
            };

            legal_with_llvm_ret_type( core::LazyOp{} );
            legal_with_llvm_ret_type( core::BinLAndOp{} );
            legal_with_llvm_ret_type( core::BinLOrOp{} );
            legal_with_llvm_ret_type( hl::ValueYieldOp{} );
```